City-of-Helsinki/drupal-hdbt
# UHF-7794 Accordion

# [UHF-7794](https://helsinkisolutionoffice.atlassian.net/browse/UHF-7794)
Created javascript library to replace the handorgel-accordion library

## What was done
- Basic accordion functionality
  - open, close, toggle
- Handle anchor tags
  - If url has anchor tag, correct accordion item is set as open
  - Handle scrolling into the anchor item automatically
- Handle focus change on element close
  - set focus on an element which is visible since close element will be hidden.
- Handle unique id creation
  - If multiple accordion items across multiple accordions have same id
    - add suffix to the element.  

## How to install

* Make sure your instance is up and running on latest dev branch.
  * `git pull origin dev`
  * `make fresh`
* Update the HDBT theme
  * `composer require drupal/hdbt:dev-UHF-7794_accordion`
  * `composer require drupal/helfi_platform_config:dev-UHF-7794_accordion`
* Run `make drush-cr`

## How to test
- Go to a page with with accordion. preferably multiple accordions.
- Open, close and toggle should work as intended
  - Element's aria-expanded should change accordingly 
- Refresh the page with anchor tag
  - Correct accordion item should be opened automatically
  - Automatic scroll after page has loaded   
- Unique id creation
  - Add an accordion with 2-3 items with same heading
  - The items should have unique id: for example "heading", "heading_1", "heading_2"
  - You can try this with 2 separate accordions on same page, it should update all duplicates

## Other PRs
https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/432


[UHF-7794]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-7794?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
